src/Compilers/Core/Portable/PublicAPI.Unshipped.txt

```diff
@@ -1,8 +1,12 @@
+abstract Microsoft.CodeAnalysis.SemanticModel.GetPreprocessingSymbolInfoCore(Microsoft.CodeAnalysis.SyntaxToken token) -> Microsoft.CodeAnalysis.PreprocessingSymbolInfo
+Microsoft.CodeAnalysis.Compilation.CreatePreprocessingSymbol(string! name) -> Microsoft.CodeAnalysis.IPreprocessingSymbol!
+Microsoft.CodeAnalysis.SemanticModel.GetPreprocessingSymbolInfo(Microsoft.CodeAnalysis.SyntaxToken token) -> Microsoft.CodeAnalysis.PreprocessingSymbolInfo
 Microsoft.CodeAnalysis.OperationKind.Attribute = 125 -> Microsoft.CodeAnalysis.OperationKind
 Microsoft.CodeAnalysis.Operations.IAttributeOperation
 Microsoft.CodeAnalysis.Operations.IAttributeOperation.Operation.get -> Microsoft.CodeAnalysis.IOperation!
 Microsoft.CodeAnalysis.SymbolDisplayLocalOptions.IncludeModifiers = 4 -> Microsoft.CodeAnalysis.SymbolDisplayLocalOptions
 Microsoft.CodeAnalysis.SymbolDisplayParameterOptions.IncludeModifiers = 2 -> Microsoft.CodeAnalysis.SymbolDisplayParameterOptions
+Microsoft.CodeAnalysis.SymbolDisplayPartKind.PreprocessingName = 33 -> Microsoft.CodeAnalysis.SymbolDisplayPartKind
```

Comment:
this is ok.  but it's unclear why this is needed at all.  what part of the process ends up using this?

Reply:
This is when creating the display parts of the preprocessing symbol, which was previously not supported by the respective SymbolDisplay implementation.

Reply:
But when/where does this happen?  Why does FAR need this?

Reply:
The LSP test failed without it, thought it was a necessary addition

Reply:
still don't understand why this is needed.  stating a test fails isn't sufficient.  Please explain why/how it failed.

Reply:
The new display part kind was added for consistency by specifying the symbol kind that is being transmitted over the LSP. The other option as I'm currently reviewing would be `Text`, though it's generic and does not specify anything. Overall, this basically adds better support for communicating the results of a FAR operation over the LSP when involving preprocessing symbols, and also works with the new classification that was added for preprocessing symbols. For better or for worse, the introduction of those APIs is a step towards enhancing the experience with preprocessing symbols, I feel it's more like a nice touch in the PR than a necessity; as I said, just `Text` could work, although it felt ugly to me.

Reply:
This doesn't sound like a test will fail. 

So I would not do this change. If you do want to do this part, please open an API request proposal. That will have to be approved before this part of the pr can go through.

Reply:
There already is an open API proposal, it will have to be included there. As for whether the test would fail, it seems that it wouldn't fail otherwise if I base it around `Text`.

Both this PR and the API proposal appear to be somewhat linked, and will be adjusted accordingly.

Reply:
If you want to link them, that's up to you. No clue when this change will go in then though.

Reply:
My personal preference is to get this merged asap with minimal overhead. Seeing that the only public APIs left are the ones involving this, I think that temporarily using the `Text` classification suffices. The motivation is to officially support classification for preprocessing symbols, which I will make in another issue, in spite of having supported FAR for them, once this PR is merged (assuming we agree on using the `Text` classification).

Reply:
I have no idea what the classification is for here.

Reply:
I also introduced a new classification for a preprocessing name specifically (see `TaggedText.cs` and the alike), which is based on the addition of this symbol display kid. Should we resort to using `SymbolDisplayPartKind.Text`, there's the appropriate mapping (`TextTags.Text`).